tests/qos/test_buffer.py

```diff
@@ -2420,19 +2456,33 @@ def _check_port_buffer_info_and_return(duthost, table, ids, port, expected_profi
         speed = port_config['speed']
         expected_profile = make_expected_profile_name(speed, cable_length, number_of_lanes=len(port_config['lanes'].split(',')))
 
+        if is_port_with_4_lossless_queues(duthost, port, tbinfo):
+            key_name = KEY_4_LOSSLESS_QUEUE
+        else:
+            key_name = KEY_2_LOSSLESS_QUEUE
         # The last item in the check list various according to port's admin state.
         # We need to append it according to the port each time. Pop the last item first
         if port_config.get('admin_status') == 'up':
             admin_up_ports.add(port)
-            buffer_items_to_check = buffer_items_to_check_dict["up"]
-            buffer_items_to_check[-1] = ('BUFFER_PG_TABLE', '3-4', profile_wrapper.format(expected_profile))
+            buffer_items_to_check = buffer_items_to_check_dict["up"][key_name][:]
+            if key_name == KEY_4_LOSSLESS_QUEUE:
+                buffer_items_to_check.extend(
+                    [('BUFFER_PG_TABLE', '2-4', profile_wrapper.format(expected_profile)),
+                    ('BUFFER_PG_TABLE', '6', profile_wrapper.format(expected_profile))])
+            else:
+                buffer_items_to_check.append(('BUFFER_PG_TABLE', '3-4', profile_wrapper.format(expected_profile)))
         else:
             if is_mellanox_device(duthost):
-                buffer_items_to_check = buffer_items_to_check_dict["down"]
-            elif is_broadcom_device(duthost) and (asic_type in ['td2'] or speed <= '10000'):
+                buffer_items_to_check = buffer_items_to_check_dict["down"][key_name]
+            elif is_broadcom_device(duthost) and (asic_type in ['td2', 'td3'] or speed <= '10000'):
```

Comment:
td3 should not be in the list. td2 is a special case where we don't apply profiles to lossless pgs for admin down ports

Reply:
Yes. This is expected and a must for reclaiming buffer.